### DISCUSSION TITLE
injectEndpoints does not have tags? [edited]

### DISCUSSION THREAD

**Discussion by samislam** [reduxjs/redux-toolkit#2568] [Q&A] [answered]
I have the following code:

```js
const roomSlice = mainApi.injectEndpoints({
  tags: ['rooms'],
  endpoints: (builder) => ({
    getMany: builder.query({
      query: () => ({
        url: `/rooms`,
        providesTags: ['rooms'],
      }),
    }),
    createOne: builder.mutation({
      query: (payload) => ({
        url: '/rooms',
        method: 'POST',
        body: payload,
        invalidatesTags: ['rooms'],
      }),
    }),
  }),
})
```

I noticed that the tags aren't doing anything, I create a *room* and then I look at the table of the rooms on my webpage, and it shows out nothing is updated until I refresh the page. 

Is it allowed to have tags inside injectEndpoints? if not, what are the alternatives? I have a big back-end API, and I need to split the code into multiple files somehow.

I found another question similar to this one, but it doesn't talk about
 invalidatesTags: ['rooms'],
and 
providesTags: ['rooms'],
and it does not give a fulfilling answer

**Comment by phryneas** [comment, marked as answer]
you got the nesting wrong and there is no plain "tags" option


```diff
const roomSlice = mainApi
+.enhanceEndpoints({
+  addTagTypes: ['rooms'],
+})
.injectEndpoints({
-  tags: ['rooms'],
  endpoints: (builder) => ({
    getMany: builder.query({
      query: () => ({
        url: `/rooms`,
-       providesTags: ['rooms'],
      }),
+    providesTags: ['rooms'],
    }),
    createOne: builder.mutation({
      query: (payload) => ({
        url: '/rooms',
        method: 'POST',
        body: payload,
-      invalidatesTags: ['rooms'],
      }),
+    invalidatesTags: ['rooms'],
    }),
  }),
})
```

**Reply by samislam** [reply]
so it means that it's actually allowed, right? just the addTagTypes, this one must be defined inside enhanceEndpoints

**Reply by samislam** [reply]
OK, is it allowed to have the addTagTypes defined inside the `mainApi` slice?

**Reply by phryneas** [reply]
it could also be in mainApi. The main problem is that you got the nesting all wrong and sent it as an option to your baseQuery instead of it being an option to `builder.query`

**Reply by samislam** [reply]
@phryneas Thank you
